Add BQ dataset's partition expiry (#6015)
segmentio/segment-docs: src/connections/storage/catalog/bigquery/index.md

### I see duplicates in my tables.

This behavior is expected. Segment only de-duplicates data in your views. Refer to the [schema section](#schema) for more details.

> ### BigQuery Default Partition Expiration

Reviewer: ```suggestion
### Why does some of my older BigQuery data expire? 
```

Reviewer: I formatted this as a question to keep parity with the other questions on this page!

> If you notice that your BigQuery data is getting deleted after a specific period of time, then it might be due to a [dataset's default table expiration](https://cloud.google.com/bigquery/docs/updating-datasets#partition-expiration) in BigQuery that sets a 90-day (or similar) expiration on all partitioned tables that are created. 

Reviewer: ```suggestion
If you notice that you are missing older BigQuery data, it might be due to a [dataset's default table expiration](https://cloud.google.com/bigquery/docs/updating-datasets#partition-expiration){:target="_blank”} in BigQuery. The default table expiration sets a standard expiration on all partitioned tables that are created. 
```

> However, you can safely remove these expirations from the tables/dataset and change them to ‘Never’, and change the dataset's default table expiration as needed. We will then be able to run a backfill for you to send all the historical data to your warehouse.

Reviewer: ```suggestion
You can safely change the default table expiration to ‘Never’, which removes these expirations from the tables/dataset and changes the dataset's default table expiration. Segment can then run a backfill for you and send all of your historical data to your warehouse.
```

Reviewer: Does a user have to request a backfill or is one initiated automatically?

Author: The customer has to request a backfill, it is not initiated automatically.
